Add notes for meeting # 28 (#60)
ethereum-cat-herders/EIPIP: All EIPIP Meetings/Meeting 028.md

# EIP# EIP IP Meeting 28 Notes
## Meeting Date/Time: Wednesday 10 March at 15:00 UTC
### Meeting Duration: 1 hour
### [GitHub Agenda Page](https://github.com/ethereum-cat-herders/EIPIP/issues/56)
### [Audio/Video of the meeting](https://www.youtube.com/watch?v=fKQVb-s0Tzs)
### Moderator: Pooja Ranjan
### Notes: Jim Bennett

----
## DECISION ITEMS

**DECISION 28.1**: Continue to think about how to deal with suggested changes in Last Call and keep Item #2 on the agenda for the next meeting.[see 20:23](https://youtu.be/fKQVb-s0Tzs?t=1219)

**DECISION 28.2**: Put an item on the next meeting agenda about EIPs for Berlin moving to Last Call. [see 32:39](https://youtu.be/fKQVb-s0Tzs?t=1959)

**DECISION 28.3**: Discuss separating ERCs from the EIP repo in the next call. [see 39:53](https://youtu.be/fKQVb-s0Tzs?t=1959)
----
**Pooja Ranjan**
Welcome to EIPIP meeting 28. I'm Pooja Ranjan.

## AGENDA

## 1. [Progress on EIP GitHub repo action bot](https://youtu.be/fKQVb-s0Tzs?t=4)

> The first item on the agenda is progress on EIP GitHub repo action bot. So a little more from the ECH team is working on the EIP GitHub repo. And I suppose she also created a pull requests to share this bot for review with the EIP editors. The pull request number is 3351. If there is any other update on the progress and concern with the bot that anyone would like to share in development of this effort, feel free to share, because I don't see any data here. But I have seen some comments left by the editors on the issue.

Reviewer: ```suggestion
The first item on the agenda is progress on EIP GitHub repo action bot. So, Alita Moore from the ECH team is working on the EIP GitHub repo. And I suppose she also created a pull requests to share this bot for review with the EIP editors. The pull request number is 3351. If there is any other update on the progress and concern with the bot that anyone would like to share in development of this effort, feel free to share, because I don't see Alita here. But I have seen some comments left by the editors on the issue.
```

**Hudson Jameson**
I would lean toward this to help with the fact that to basically stick with somewhat of the status quo, but to try to also push best practices would be to keep Last Call's definition as is, pretty much, because it doesn't include all the security vulnerabilities right now and move it to DevNet phase. And then public testnet doesn't include an EIP status beyond... but then that leaves public testnets to not have an EIP status. That's the problem, I guess. Yeah.

**Pooja Ranjan**

> One more thing I would like to add here is generally, the definition of Last Call for that is [unclear](https://youtu.be/fKQVb-s0Tzs?t=1128). So one correction there would be a minimum of 14 days would be good. But you know, if they want to keep it in Last Call for over a month, because when it is on testnet, they believe it is like [unclear](https://youtu.be/fKQVb-s0Tzs?t=1139). So I am not very sure that we should put the Last Call status in DevNet phase.

Reviewer: ```suggestion
One more thing I would like to add here is generally, the definition of Last Call suggests it stays there for 14 days. So one correction there would be "a minimum of 14 days" would be good. But you know, if we want to keep it in Last Call for over a month, because when it is on testnet, I believe it is like four weeks at least. So I am not very sure that we should put the Last Call status in the DevNet phase.
```

> 3

Reviewer: ```suggestion

```

**Hudson Jameson**
It's more the discussion links. But we wouldn't be deleting PRs or issues, would we?

**Micah**
No. We don't have the ability to do them.

**Hudson Jameson**
Got it. Okay. Sounds good. I've got to jump off. I'll think about that part more. Thanks, everyone.

**Edson Allyon**
Appreciate it. Alita's here. Did you want to go back to the bot?

**Pooja Ranjan**
Yeah.

**lightclient**
One last thing on that real quick. Theoretically if we did want to do that, does anyone have a theoretical path to making that happen? Who needs to agree to that to make something like that happen?

**Micah**
The benevolent dictator of EIPs.

**lightclient**
So we're gonna do it tomorrow then.

**Micah**
Yeah, I don't know. So it would be people who actually look at ERCs are me and, in theory, more EIP editors that are very few and far between, and then there are the authors of those ERCs themselves. And then once the ERCs get popular, there are the people who are looking at them to implement. Those are the three people I have seen show up in the EIPs repo related to ERCs. These are the three types of people.

**Pooja Ranjan**

> I have been looking at it recently just only for one thing - if there is a status review or they are trying to escape it right from Draft the last column. That's my only comment for any ERC that I have logged into so far. Maybe this is something that we could bring it on the next call and as an item itself. Do we want to separate the EIP and ERC, if people are open to discuss that.

Reviewer: ```suggestion
I have been looking at it recently just only for one thing - if there is a status review or they are trying to escape it right from Draft the last call. That's my only comment for any ERC that I have looked into so far. Maybe this is something that we could bring it on the next call and as an item itself. Do we want to separate the EIP and ERC, if people are open to discuss that.
```

**Micah**
So my feeling is that one, two weeks aligns with where my gut was, just kind of looking over your shoulder. That's what the order of magnitude is. This is almost a question of should you test your test sweep? Because the bot is effectively just a kind of a test runner that is supposed to building an automated CI process? and spending one to two weeks? We can definitely test your tests. It's just a question of is it worth the time. The time be better spent making the bot actually do better, making it do more things. There's a list of things we'd rather have the bot do. We have one or two weeks of engineering time. We can either sink it on making sure that the bot is robust, or we can sink it on making the bot more feature rich, and I'm on the fence as to which is more valuable.

> 3

Reviewer: ```suggestion

```

**lightclient**
Theoretically, there's definitely ways where you can tag a bot and give it a command. I know Boris has a lot of functionality like that, where you can tag Boris and say, "Okay, once you know the other people on this approve, go ahead and merge it."" I don't know how hard it is to implement functionality really simply. But it can be done.

**Alita**
Because we're short on time, I wanted to hop on one other thing. Did you all want me to go out and build the botinto a separate repo, and I can just do it, I can set it up, and then Hudson or whoever has permissions, I can just give over control or something like that?

> M

Reviewer: ```suggestion

```

**Micah**
I'm on the fence. Lightclient half convinced me. The plus side of having it in the EIPs repo is that it's right there. And the same people who have write access to the EIPs repo would have write access to the bot. So when we, the people who have write access to that repo, notice it doing something wrong, or we want to make a quick patch, we know we have permissions. And so if we add new editors, those editors will automatically get the access rights. Having a separate repo is just like a modularity and keeping separation of concerns. That doesn't make more sense. In that regard, I'm on the fence. If other people have really strong problems, it's definitely one down on either side.

> 3

Reviewer: ```suggestion

```

**Alita**
All right. Sounds good.

**Pooja Ranjan**
Thank you Alita, Micah, and lightclient discussion, it seems like we are going to see some more actions. And we would want to keep this in the next meeting item as well. So we'll wait for your progress there. I think we are over time. I wanted to quickly cover on the EIP editors onboarding thing. We just have tried to answer most of the questions on the EIP Discord itself. But if you could not get answer, I have created this new role in the ECH Discord as EIP editors aspirants, and if you are not there, you want to be on part of there, if you have any further questions, please reach out to us. You can DM to any of the members of the of the Ethereum Cat Herders or post your question directly in the general section of the ECH Discord. That was the last item on the agenda. Thank you, everyone, for joining us today.

See you all and still on March 24. Thank you.

# Attendees

* Alita
* Brent Allsop
* Edson Ayllon
* Hudson Jameson
* Jim Bennett
* lightclient
* Micah Zoltu
* Pooja Ranjan (Host)

> # Date for Next Meeting: November 18 at 1500 UTC

Reviewer: ```suggestion
# Date for Next Meeting: March 24 at 1500 UTC
```
